{
    "sourceFile": "node_modules/es-abstract/2022/AsyncIteratorClose.js",
    "activeCommit": 0,
    "commits": [
        {
            "activePatchIndex": 0,
            "patches": [
                {
                    "date": 1746892149061,
                    "content": "Index: \n===================================================================\n--- \n+++ \n"
                }
            ],
            "date": 1746891704042,
            "name": "cart",
            "content": "'use strict';\n\nvar GetIntrinsic = require('get-intrinsic');\n\nvar $SyntaxError = require('es-errors/syntax');\nvar $TypeError = require('es-errors/type');\nvar $Promise = GetIntrinsic('%Promise%', true);\n\nvar Call = require('./Call');\nvar CompletionRecord = require('./CompletionRecord');\nvar GetMethod = require('./GetMethod');\n\nvar isObject = require('../helpers/isObject');\nvar isIteratorRecord = require('../helpers/records/iterator-record-2023');\n\nvar callBound = require('call-bound');\n\nvar $then = callBound('Promise.prototype.then', true);\n\n// https://262.ecma-international.org/12.0/#sec-asynciteratorclose\n\nmodule.exports = function AsyncIteratorClose(iteratorRecord, completion) {\n\tif (!isIteratorRecord(iteratorRecord)) {\n\t\tthrow new $TypeError('Assertion failed: `iteratorRecord` must be an Iterator Record'); // step 1\n\t}\n\n\tif (!(completion instanceof CompletionRecord)) {\n\t\tthrow new $TypeError('Assertion failed: completion is not a Completion Record instance'); // step 2\n\t}\n\n\tif (!$then) {\n\t\tthrow new $SyntaxError('This environment does not support Promises.');\n\t}\n\n\tvar iterator = iteratorRecord['[[Iterator]]']; // step 3\n\n\treturn $then(\n\t\t$then(\n\t\t\t$then(\n\t\t\t\tnew $Promise(function (resolve) {\n\t\t\t\t\tresolve(GetMethod(iterator, 'return')); // step 4\n\t\t\t\t\t// resolve(Call(ret, iterator, [])); // step 6\n\t\t\t\t}),\n\t\t\t\tfunction (returnV) { // step 5.a\n\t\t\t\t\tif (typeof returnV === 'undefined') {\n\t\t\t\t\t\treturn completion; // step 5.b\n\t\t\t\t\t}\n\t\t\t\t\treturn Call(returnV, iterator); // step 5.c, 5.d.\n\t\t\t\t}\n\t\t\t),\n\t\t\tnull,\n\t\t\tfunction (e) {\n\t\t\t\tif (completion.type() === 'throw') {\n\t\t\t\t\tcompletion['?'](); // step 6\n\t\t\t\t} else {\n\t\t\t\t\tthrow e; // step 7\n\t\t\t\t}\n\t\t\t}\n\t\t),\n\t\tfunction (innerResult) { // step 8\n\t\t\tif (completion.type() === 'throw') {\n\t\t\t\tcompletion['?'](); // step 6\n\t\t\t}\n\t\t\tif (!isObject(innerResult)) {\n\t\t\t\tthrow new $TypeError('`innerResult` must be an Object'); // step 10\n\t\t\t}\n\t\t\treturn completion;\n\t\t}\n\t);\n};\n"
        }
    ]
}